### ISSUE TITLE
Default sphinx role "autolink" appears to be deprecated

### ISSUE THREAD

**Issue by timhoffm** [numpy/numpy#18879]
I noticed that numpy uses a shpinx config setting

https://github.com/numpy/numpy/blob/05b12102f93f18c0a8bbd86aa2e4a6fcbb554ab4/doc/source/conf.py#L125

This is poor documented at sphinx, but seems to be deprecated:
https://www.sphinx-doc.org/en/master/search.html?q=autolink

and is scheduled for removal in Sphinx 4.0.

**Comment by Ciozzz** [comment]
Hi Tim,
According to [this discussion on scikit-learn](https://github.com/scikit-learn/scikit-learn/issues/12535), I think it better to change `autolink` to another value, like `any`, so `default_role` can become more useful.

**Comment by timhoffm** [comment]
I agree that the default role needs changing (if only for the fact that "autolink" seems deprecated). At Matplotlib, we use `obj`, which is a bit more restrictive than `any`. Most things we link are other classes/functions. Also trying to catch `ref`, `doc` and `option` seemed a little too ambitious; but YMMV.

Note that when changing to `any` or `obj`, you'll get warnings for the current numpydoc style which uses the default role for parameters (see https://github.com/numpy/numpydoc/issues/323). That would have to be addressed one way or the other.
